pkg/ocserv/ocserv.html

```diff
@@ -0,0 +1,173 @@
+the Free Software Foundation; either version 2.1 of the License, or
+(at your option) any later version.
+
+Cockpit is distributed in the hope that it will be useful, but
+WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License
+along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
+-->
+<html>
+
+<head>
+    <title>OpenConnect VPN</title>
+    <meta charset="utf-8">
+    <link href="../base1/cockpit.css" type="text/css" rel="stylesheet">
+    <link href="ocserv.css" type="text/css" rel="stylesheet">
+    <script type="text/javascript" src="../base1/cockpit.js"></script>
+</head>
+<body>
+
+    <div id="status-info" class="container-fluid">
+	<div class="panel panel-default">
+		<div class="panel-heading">
+			<span id="head">Server status</span>
+		</div>
+		<table class="cockpit-info-table" style="margin-bottom:10px"
+		id="status_table">
+		</table>
+	</div>
+    </div>
+
+    <div id="users-info" class="container-fluid">
+	<div class="panel panel-default">
+		<div class="panel-heading">
+			<span id="head">Connected users</span>
+		</div>
+		<table class="table table-hover" style="margin-bottom:10px"
+		id="users_table">
+	        <thead>
+	        <tr>
+	          <th class="ocserv-users-id">ID</th>
+  	          <th class="ocserv-users-name">Username</th>
+	          <th class="ocserv-users-remote-ip">Remote IP</th>
+	          <th class="ocserv-users-ip">VPN IP</th>
+		  <th class="ocserv-users-since">Since</th>
+		  <th class="ocserv-users-state">State</th>
+		</tr>
+	        </thead>
+	        <tbody id="users_table_tbody">
+
+	        </tbody>
+		</table>
+	</div>
+    </div>
+
+
+<script>
+require([
+    'jquery',
+    'base1/cockpit'
+], function($, cockpit) {
+        var raw;
+        var status_raw;
+
+        occtl_users_run();
+        occtl_status_run();
+
+        function occtl_users_run() {
+            var proc = cockpit.spawn(["/usr/bin/occtl", "-n", "-j", "show", "users"]);
+            proc.done(occtl_users_success);
+            proc.stream(occtl_users_output);
+            proc.fail(occtl_users_fail);
+
+            raw = "";
+        }
+
+        function occtl_users_success() {
+            var obj;
+            try {
+            	obj = JSON.parse(raw);
+	    } catch(e) {
+	    	console.warn(e, obj);
+	    }
+
+	    $("#users_table_tbody").empty();
+
+	    $.each(obj, function (index,Object) {
+	    	var ipv4 = Object.IPv4;
+	    	var ipv6 = Object.IPv6;
+	    	var ip;
+
+	    	if (ipv4 && ipv6) {
+		    ip = ipv4 + "/" + ipv6;
+		} else {
+		    if (ipv4)
+			ip = ipv4;
+		    else
+			ip = ipv6;
+		}
+
+	    	$("#users_table").append('<tr><td class="ocserv-users-id">' + Object.ID + '</td>' +
+		    '<td class="ocserv-users-name">' + Object.Username + '</td>' + 
```

Comment:
This is a XSS security vulnerability. A user could have HTML in their name, and it would be interpretted here, including scripts and javascript. Please use Mustache instead or use jQuery text() function to prevent parsing data as HTML.